src/application/listeners/projectEventListeners.ts

```diff
@@ -1,81 +1,99 @@
 
-import { postRemoteProject } from '../../api/projectsAPI';
+import { createRemoteProject, ProjectInfo, readAllProjects, readProjectById } from '../../api/projectsAPI';
 import { createFileAtom } from '../../atoms/fileEntryActions';
 import {
   closeProjectAtom,
   isProjectOpenAtom,
   newProjectAtom,
   newSampleProjectAtom,
-  newWebProjectAtom,
   openProjectAtom,
-  openWebProjectAtom,
 } from '../../atoms/projectState';
 import { getNewProjectsBaseDir } from '../../io/filesystem';
-import { notifyInfo } from '../../notifications/notifications';
+import { notifyInfo, notifyWarning } from '../../notifications/notifications';
 import { saveCachedSettings, setCachedSetting } from '../../settings/settingsManager';
-import { readOneProjectFromWeb } from '../../web';
+
+export const SAMPLE_PROJECT_NAME = 'RefStudio Sample';
 
 export function useFileProjectNewListener() {
   const newProject = useSetAtom(newProjectAtom);
-  const newWebProject = useSetAtom(newWebProjectAtom);
   const createFile = useSetAtom(createFileAtom);
 
   return async () => {
+    let projectInfo: ProjectInfo;
     if (import.meta.env.VITE_IS_WEB) {
-      // Open a custom dialog to ask the project name
-      const projectInfo = await postRemoteProject('Project Web');
-      persistProjectDirInSettings(projectInfo.projectPath);
-      await newWebProject(projectInfo.projectId, projectInfo.projectPath, projectInfo.projectName);
-      createFile();
-      notifyInfo('New project created at ' + projectInfo.projectPath);
+      const projectName = 'Project Web'; // TODO: Open a custom dialog to ask the project name
+      projectInfo = await createRemoteProject(projectName);
     } else {
       const newProjectPath = await save({ defaultPath: await getNewProjectsBaseDir() });
-      if (typeof newProjectPath === 'string') {
-        await newProject(newProjectPath);
-        createFile();
-        persistProjectDirInSettings(newProjectPath);
-        notifyInfo('New project created at ' + newProjectPath);
+      if (typeof newProjectPath !== 'string') {
+        notifyInfo('User canceled operation to create new project');
+        return;
       }
+      const projectName = newProjectPath.split('/').pop() ?? 'Project';
+      projectInfo = await createRemoteProject(projectName, newProjectPath);
     }
+
+    await newProject(projectInfo.id, projectInfo.path, projectInfo.name);
+    createFile();
+    notifyInfo('New project created with success');
+    persistProjectIdInSettings(projectInfo.id);
   };
 }
 
 export function useFileProjectNewSampleListener() {
   const sampleProject = useSetAtom(newSampleProjectAtom);
+  const createFile = useSetAtom(createFileAtom);
   return async () => {
-    const newProjectPath = `${await getNewProjectsBaseDir()}RefStudio Sample`;
-    await sampleProject(newProjectPath);
-    persistProjectDirInSettings(newProjectPath);
-    notifyInfo('New project created at ' + newProjectPath);
+    const projects = await readAllProjects();
```

Comment:
should this logic (create or return sample project) live in the backend?

Reply:
Most likely yes, with a speacialized API for sample projects.